### PR TITLE
update used interface

### PR DESCRIPTION
**Pull request by Neil-Cao**

### CODE REVIEW

File: EaseUI/EMUIKit/ViewController/EaseConversationListViewController.m [hunk 1]
```diff
@@ -172,39 +172,42 @@ - (void)tableViewDidTriggerHeaderRefresh
...
-                    model = [weakself.dataSource conversationListViewController:weakself
-                                                       modelForConversation:converstion];
-                }
-                else{
-                    model = [[EaseConversationModel alloc] initWithConversation:converstion];
-                }
-                
-                if (model) {
-                    [weakself.dataArray addObject:model];
+            EaseConversationListViewController *strongSelf = weakself;
```
Comment (jerrywu12) [inline EaseUI/EMUIKit/ViewController/EaseConversationListViewController.m:175]: won't this be kinda odd, pointing to weakself but called strongself? why not just use self? 
self.dataSource, self.tableView, etc.

File: EaseUI/EMUIKit/ViewController/EaseMessageViewController.m [hunk 2]
```diff
@@ -206,53 +206,60 @@ - (void)joinChatroom:(NSString *)chatroomId
...
-            [strongSelf hideHud];
-            [strongSelf showHint:[NSString stringWithFormat:NSLocalizedString(@"chatroom.joinFailed",@"join chatroom \'%@\' failed"), chatroomId]];
-        } else {
-            if (aError.code == EMErrorChatroomAlreadyJoined) {
-                [[EMClient sharedClient].roomManager asyncLeaveChatroom:chatroomId success:^(EMChatroom *aRoom) {
-                    [[EMClient sharedClient].chatManager deleteConversation:chatroomId deleteMessages:YES];
-                } failure:^(EMError *aError) {
-                    //leave failed
+    [[EMClient sharedClient].roomManager joinChatroom:chatroomId completion:^(EMChatroom *aChatroom, EMError *aError) {
+        EaseMessageViewController *strongSelf = weakSelf;
```
Comment (jerrywu12) [inline EaseUI/EMUIKit/ViewController/EaseMessageViewController.m:210]: use self?

File: EaseUI/EMUIKit/ViewController/EaseMessageViewController.m [hunk 3]
```diff
@@ -744,70 +750,79 @@ - (void)loadMessagesBefore:(NSString*)messageId
...
-                            *stop = YES;
+    
+    void (^refresh)(NSArray *messages) = ^(NSArray *messages) {
+        dispatch_async(_messageQueue, ^{
+            //Format the message
+            NSArray *formattedMessages = [weakSelf formatMessages:messages];
+            
+            //Refresh the page
+            dispatch_async(dispatch_get_main_queue(), ^{
+                EaseMessageViewController *strongSelf = weakSelf;
```
Comment (jerrywu12) [inline EaseUI/EMUIKit/ViewController/EaseMessageViewController.m:761]: use self instead of creating strongSelf?

**Comment by jerrywu12** [comment]
@Neil-Cao only question about using "self" instead of "strongSelf". everything else looks good!

File: EaseUI/EMUIKit/ViewController/EaseConversationListViewController.m (hunk 1 above)
Comment (Neil-Cao) [inline EaseUI/EMUIKit/ViewController/EaseConversationListViewController.m:175]: 一般来说UI进行异步处理时，异步block里面使用weak修饰的变量，避免block执行完之前UI无法释放的问题

File: EaseUI/EMUIKit/ViewController/EaseMessageViewController.m (hunk 2 above)
Comment (Neil-Cao) [inline EaseUI/EMUIKit/ViewController/EaseMessageViewController.m:210]: 同上

File: EaseUI/EMUIKit/ViewController/EaseMessageViewController.m (hunk 3 above)
Comment (Neil-Cao) [inline EaseUI/EMUIKit/ViewController/EaseMessageViewController.m:761]: block 会retain它用到的外部变量，block必须等到异步任务结束才会释放，所以加weak/strong转换避免ui在任务执行完之前release不了的问题

Comment (jerrywu12) [inline EaseUI/EMUIKit/ViewController/EaseMessageViewController.m:761]: ok

File: EaseUI/EMUIKit/ViewController/EaseMessageViewController.m (hunk 2 above)
Comment (jerrywu12) [inline EaseUI/EMUIKit/ViewController/EaseMessageViewController.m:210]: ok

File: EaseUI/EMUIKit/ViewController/EaseConversationListViewController.m (hunk 1 above)
Comment (jerrywu12) [inline EaseUI/EMUIKit/ViewController/EaseConversationListViewController.m:175]: ok, thanks.

**Comment by jerrywu12** [comment]
Ready to merge if SDK and demo are ready.

**Pull request merged**